pkg/vere/main.c

```diff
@@ -1960,55 +1962,168 @@ _cw_pack(c3_i argc, c3_c* argv[])
   u3m_stop();
 }
 
-/* _cw_play(): replay events.
+/* _cw_play_slog(): print during replay.
+*/
+static void
+_cw_play_slog(u3_noun hod)
+{
+  u3_pier_tank(0, 0, u3k(u3t(hod)));
+  u3z(hod);
+}
+
+/* _cw_play_exit(): exit immediately.
+*/
+static void
+_cw_play_exit(c3_i int_i)
+{
+  //  explicit fprintf to avoid allocation in u3l_log
+  //
+  fprintf(stderr, "\r\n[received keyboard stop signal, exiting]\r\n");
+  raise(SIGINT);
+}
+
+/* _cw_play(): replay events, but better.
 */
 static void
 _cw_play(c3_i argc, c3_c* argv[])
 {
-  c3_i ch_i, lid_i;
+  c3_i lid_i, ch_i;
   c3_w arg_w;
+  c3_o ful_o = c3n;
+  c3_o mel_o = c3n;
+  c3_o sof_o = c3n;
+  c3_d eve_d = 0;
+  c3_d sap_d = 0;
 
-  static const c3_c usage_c[] = "error: invalid usage, expected "
-                                "`urbit play [--replay-to <event_num> "
-                                "| --batch-size <event_cnt>] <pier>`";
   static struct option lop_u[] = {
-    { "batch-size", required_argument, NULL, 'b' },
-    { "replay-to",  required_argument, NULL, 'n' },
-    { NULL,         0,                 NULL, 0   }
+    { "loom",      required_argument, NULL, c3__loom },
+    { "no-demand", no_argument,       NULL, 6 },
+    { "auto-meld", no_argument,       NULL, 7 },
+    { "soft-mugs", no_argument,       NULL, 8 },
+    { "full",      required_argument, NULL, 'f' },
```

Comment:
Good catch, thanks. I missed this because I was testing with `urbit play -f $pier`, where `$pier` was doubling up as the argument to `-f`, then ignored.